test/node/regress-gh-329.js

```diff
@@ -0,0 +1,47 @@
+// Regression test for https://github.com/metarhia/jstp/pull/329
+
+'use strict';
+
+const test = require('tap');
+const jstp = require('../..');
+
+const app = new jstp.Application('app', {});
+let server;
+
+let inspectSent = false;
+
+test.beforeEach((done) => {
+  server = jstp.net.createServer([app]);
+  server.on('log', (connection, event) => {
+    if (event === 'inspect') {
+      inspectSent = true;
+    }
+  });
+  server.listen(0, done);
+});
+
+test.afterEach((done) => {
+  server.close();
+  done();
+});
+
+test.test('connectAndInspect must accept empty array of interfaces', (test) => {
+  test.plan(4);
+  jstp.net.connectAndInspect(
+    app.name,
+    null,
+    [],
+    server.address().port,
+    (error, connection, api) => {
+      test.pass('must invoke the callback');
+      test.assertNot(error, 'must connect to server');
+      test.equal(
+        Object.keys(api).length,
+        0,
+        'remote proxy object must be empty'
```

Comment:
This may be misleading, as `api` is an object that contains `RemoteProxy`, but `remote proxy object` may be interpreted as 'instance of `RemoteProxy`'.